### PR TITLE
Feature/associations

### PR DESCRIPTION
**Pull request by franciscoPonceDev**
### Project requirements
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -20,6 +22,10 @@ def correct_name
     @name
   end
 
+  def add_rental(date, person)
+    Rental.new(date, self, person)
+  end
```
Comment (Deepakdanger) [inline person.rb]: - [ ] Rental can be created with ,date, person, & book parameter.Kindly change according to the requirement.
```
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label, :students
```
Comment (Deepakdanger) [inline classroom.rb]: - [ ] Here, only `:label` should be in `attr_accessor`. Kindly move `:students` to the `attr_reader`.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (Deepakdanger) [inline book.rb]: - [ ] Here, `:rentals` should be in `attr_accessor`. Kindly fix this.

File: person.rb [hunk 4]
```diff
@@ -1,7 +1,8 @@
 require './nameable'
+require './rentals'
 
 class Person < Nameable
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (Deepakdanger) [inline person.rb]: - [x] Here, `:rentals` should be in `attr_accessor`. Kindly fix this.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @franciscoPonceDev, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kingstalux — APPROVED** [review]
Hi @franciscoPonceDev,
# Status: Approved ✔️ ✔️ ✔️ 
 
68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f78543078657a514755357843444a754350652f67697068792e676966


Great job implementing the required changes 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**